File: Algorithms & Data Structures/Data structures implementation C++/Stack.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cstdlib>
using namespace std;

struct node {
  int data;
  node *next;
};


// Stack //

class stack {
  public:
    node *top;
    stack() {
      top = NULL;
    }
    void push(int item) {
      node *temp = new node;
      temp->data = item;
      temp->next = top;
      top = temp;
    }
    int pop() {
      node *old = top;
      top = top->next;
      int oldData = old->data;
      delete old;
      return oldData;
    }
};

//Stack END //


int main() {
 // Stack: push, pop
 stack s;
}

File: Algorithms & Data Structures/Data structures implementation JS/Stack.js
class Node {
    constructor(item) {
        this.item = item;
        this.right = null; // alias for 'next'
        this.top = null;
    }
}

class Stack {
    constructor() {
        this.length = 0;
        this.top = null;
    }
    isEmpty() {
        return this.top == null;
    }
    push(item) {
        let newNode = new Node(item);
        newNode.right = this.top;
        this.top = newNode;
        this.length++;
    }
    pop() {
        if (!this.isEmpty()) {
            this.top = this.top.right;
            this.length--;
        }
        else {
            console.log("Stack is empty!");
        }
    }
    print() {
        let current = this.top;
        while (current) {
            console.log(current.item + "\n");
            current = current.right;
        }
    }
}